### PR TITLE
feat(table): update table to support error messages

### PR DESCRIPTION
**Pull request by lavanya-bmw**
## Description

Updated table component to support error messages

## Why

to enhance table component to support error messages

## Issue

https://github.com/eclipse-tractusx/portal-frontend/issues/891
https://github.com/eclipse-tractusx/portal-frontend/issues/888

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

File: src/components/basic/Table/index.tsx [hunk 1]
```diff
@@ -175,6 +176,25 @@
     }
   }
 
+  const renderErrorMessage = () => {
+    if (error == null) {
+      return <Typography variant="body2">{noRowsMsg ?? 'No rows'}</Typography>
+    } else if (error != null) {
```
[the remark this answers is not in the document]
Comment (lavanya-bmw) [reply src/components/basic/Table/index.tsx]: Removed Unneeded defensive code

File: src/components/basic/Table/index.tsx [hunk 2]
```diff
@@ -175,6 +176,25 @@ export const Table = ({
     }
   }
 
+  const renderErrorMessage = () => {
+    if (error == null) {
+      return <Typography variant="body2">{noRowsMsg ?? 'No rows'}</Typography>
+    } else {
```
Comment (oyo) [inline src/components/basic/Table/index.tsx:182]: we don't need else if the previous if returns 100% sure. Change from

```diff
- if (something) {
-   return x
- } else {
-   doMore()
- }
``` 

to

```diff
+ if (something) {
+   return x
+ }
+ doMore()
```

File: src/components/basic/Table/index.tsx [hunk 3]
```diff
@@ -175,6 +176,25 @@ export const Table = ({
...
+      if (error.status >= 400 && error.status < 500) {
+        return <Error400Overlay errorMessage4xx={error.message} />
+      } else {
```
Comment (oyo) [inline src/components/basic/Table/index.tsx:185]: again no else needed

**Review by oyo — CHANGES_REQUESTED** [review]
unnecessary else

**Review by oyo — APPROVED** [review]

**Pull request merged**